### ISSUE TITLE
need architecture page

### ISSUE THREAD

**Issue by tomthetrainer** [eclipse/deeplearning4j#3087]
Under open source in the navpanel there is link to architecture page, the link is commented out and the architecture page is long gone. I need to build a new architecture page.

**Comment by tomthetrainer** [comment]
I have no reference on what this would look like? A flow chart or graph of the pieces? Is there an example from another project that I could reference as a starting point?

**Comment by AlexDBlack** [comment]
Some initial thoughts here on what I might include, if I was doing it:
- Broad overview of functionality: 
    - How ND4J, DataVec, Arbiter etc relate to DL4J
    - For DL4J: quick summary of the functionality we do/don't have (neural networks, distributed training, NLP, UI, etc)
- DL4J: Start with an overview of the major modules. deeplearning4j-nn, deeplearning4j-core, etc
    - Short description of each module
    - Any requirements or issues (java 7 or 8? multiple scala versions? multiple spark versions?)
    - Maybe: Some examples of what dependencies to use for common use cases
- DL4J specifics
    - Major classes to be aware of: MultiLayerNetwork vs. ComputationGraph. Layers. NLP stuff?
    - Some of the design decisions: Data stuff in datavec, layer config vs. implementation split, use of JSON for serialization of most configuration
    - Custom functionality: what can we do (custom layers, activation functions, graph vertices, loss functions) and what can't we do yet (custom updaters)

**Comment by saudet** [comment]
A figure like we have for SKIL at https://skymind.ai/products, but for the core components, would look good to me.
